lunes_cms/api/v2/views/unit_words_viewset.py

```diff
@@ -0,0 +1,46 @@
+from django.db.models import Q
+from rest_framework import viewsets
+from rest_framework.exceptions import PermissionDenied
+
+from ....cmsv2.models import Unit
+from ..serializers import UnitWordRelationSerializer
+from ....cmsv2.models.unit import UnitWordRelation
+
+
+class UnitWordViewSet(viewsets.ModelViewSet):
+    """
+    Retrieve the list of all words that belong to a given unit
+    """
+
+    serializer_class = UnitWordRelationSerializer
+    http_method_names = ["get"]
+
+    def get_queryset(self):
+        """
+        Get the queryset of unit word relations
+
+        :return: The queryset of unit word relations
+        :rtype: ~django.db.models.query.QuerySet
+        """
+        if getattr(self, "swagger_fake_view", False):
+            return UnitWordRelation.objects.none()
+
+        units = Unit.objects.filter(
+            id=self.kwargs["unit_id"], released=True, jobs__released=True
+        ).distinct()
+        if len(units) != 1:
+            raise PermissionDenied()
+        unit = units[0]
+
+        queryset = (
+            UnitWordRelation.objects.select_related("word")
+            .filter(
+                unit=unit,
+                word__audio_check_status="CONFIRMED",
+            )
+            .filter(
+                Q(image_check_status="CONFIRMED")
+                | Q(word__image_check_status="CONFIRMED")
+            )
```

Comment:
Sorry, maybe the issue description was not specific enough. We only want to send out words that:
- _if there is an image for the unit<>word relation:_ that specific image should be `CONFIRMED`
- _if there is no relation specific image:_ the default image (at the word) should be `CONFIRMED`

Reply:
I think I don't get the difference. Do you mean that it is possible for a relation to have no image but still have `image_check_status` set to "CONFIRMED"?